namespaces/referencing-namespace.md

```diff
@@ -0,0 +1,194 @@
+    "id": "my-foo"
+  },
+  "content": "bar"
+}
+```
+
+In this example, the `foo` element has an `id` of `my-foo`, which can be used to reference this specific instance of the `foo` element.
+
+### Roles (Element)
+
+The `roles` element allows for giving specific roles to element based on the context in which they are found. The roles MAY be reused throughout the document to provide more specific classification, uses, and roles of elements. It provides element reuse without the need for extending or creating new elements.
+
+#### Properties
+
+- `element`: roles (string, required)
+- `content` (array[string])
+
+#### Example
+
+This example shows the `roles` element being used as an attribute.
+
+```json
+{
+  "element": "user",
+  "attributes": {
+    "roles": ["customer"]
+  },
+  "content": [
+    {
+      "element": "id",
+      "content": 4
+    }
+  ]
+}
+```
+
+Using `roles`, the `user` element is given a more specific classification as a customer.
+
+## Namespacing
+
+Namespacing provides a way to do the following in a document:
+
+- Include elements directly from other documents
+- Define a prefix for referencing elements in defined namespaces
+
+This functionality exists to provide extendability and prevent name collisions in the event that two or more namespace specify the same element name.
+
+### Require (Element)
+
+The `require` element allows for including elements from namespaces or defining prefixes for namespaces that may be used throughout the document.
+
+#### Properties
+
+- `element`: require (string, required)
+- `content` (array[Namespace])
+
+### Namespace (Element)
+
+The `namespace` element is used for defining and referencing namespaces depending on its context. When used in a `require` element, it defines namespaces and their prefixes. When used in elements like `type` (defined below), it references the specified namespace.
+
+It MAY also be used as an attribute if defined by the particular namespace or element, and its used depends on how it is defined to be used.
+
+#### Properties
+
+- `element`: namespace (string, required)
+- `attributes`
+    - `prefixAs` (string, optional) - Unique name to be used to reference this namespace throughout the document.
+- `content` (string, required) - Resolvable URI of namespace, which MAY be relative or absolute
+
+#### Example
+
+```json
+{
+  "element": "require",
+  "content": [
+    {
+      "element": "namespace",
+      "content": "http://example.com/foo"
+    },
+    {
+      "element": "namespace",
+      "attributes": {
+        "prefixAs": "bar"
+      },
+      "content": "http://example.com/bar"
+    }
+  ]
+}
+```
+
+There are two namespaces used here, one without the prefix and one with the prefix.
+
+- The first namespace defined for `http://example.com/foo` MUST have all of its elements available in the document in which it is required. If there is a conflict in element names, the namespace defined latest in the document MUST receive precedence.
```

Comment:
I still think this is complex and not quite sure what it is solving that it can't be address with using the `ref` element